### PR TITLE
Adapted the maven specific mysql dependency update to the generic one to include gradle

### LINKED ISSUE #375: mysql-connector-java should use generic ChangeDependency over maven specific one
## What version of OpenRewrite are you using?
I am using Gradle plugin 6.0.0

## How are you running OpenRewrite? 

- Gradle plugin
- [a7439c7 commit of spring-petclinic](https://github.com/spring-projects/spring-petclinic/tree/a7439c74ea718c4f59fe6c7c643c4afe59d7e718) as this is the first commit with gradle on petclinic repo

Can you share your configuration so that we can rule out any configuration issues? 

- gradle 7.4
- MAC OS

### Adapt build gradle to run rewrite: 
```groovy
plugins {
  id 'org.springframework.boot' version '2.6.1'
  id 'io.spring.dependency-management' version '1.0.11.RELEASE'
  id 'java'
  id 'org.openrewrite.rewrite' version '6.0.0'
}

rewrite {
    activeRecipe(
            'org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_0',
    )
}

//...
dependencies {
  rewrite platform('org.openrewrite.recipe:rewrite-recipe-bom:2.0.0')
  rewrite('org.openrewrite.recipe:rewrite-spring')
  ...
}
```

### Run openrewrite

Launch `gradle rewriteRun`. 
Wait for build succeeded message.

**Expected behaviour**

`mysql:mysql-connector-java` is replaced with `mysql-connector-j` as it is for [maven](https://github.com/openrewrite/rewrite-spring/blob/f0151affbff68d4c6538540ae4fe605ecabc0c2e/src/main/resources/META-INF/rewrite/spring-boot-27.yml#L54-L59)

**Actual behaviour**

It is not changed.

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
Sure, if guided properly I am always open to take "newbie" tickets!

### PR DESCRIPTION
**Pull request by Jenson3210**
## What's changed?

Adapted the maven specific mysql dependency update to the generic one to include gradle

## What's your motivation?

Fixes #375 

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've added the license header to any new files through `./gradlew licenseFormat`
- [X] I've used the IntelliJ auto-formatter on affected files
- [X] I've updated the documentation (if applicable)

### CODE REVIEW

File: src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/UpdateMysqlDriverArtifactIdTest.java [hunk 1]
```diff
@@ -106,34 +105,103 @@ void doNotPinWhenNotVersioned() {
...
+                                              }
+                                                                   
+                                              repositories {
+                                                 mavenCentral()
+                                              }
+                                                                   
+                                              dependencies {
+                                                  runtimeOnly 'mysql:mysql-connector-java:8.0.30'
+                                              }
+                                              """, """
```
Comment (kunli2) [inline src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/UpdateMysqlDriverArtifactIdTest.java:157]: stylistic, put another text block to a new line.

Comment (Jenson3210) [reply src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/UpdateMysqlDriverArtifactIdTest.java]: Moved text blocks to new lines

File: src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/UpdateMysqlDriverArtifactIdTest.java [hunk 2]
```diff
@@ -15,78 +15,77 @@
...
-
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.openrewrite.gradle.Assertions.withToolingApi;
 import static org.openrewrite.maven.Assertions.pomXml;
 
-@Issue("https://github.com/openrewrite/rewrite-spring/issues/274")
+@Issue("https://github.com/openrewrite/rewrite-spring/issues/274 & https://github.com/openrewrite/rewrite-spring/issues/375")
```
[the remark this answers is not in the document]
Comment (Jenson3210) [reply src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/UpdateMysqlDriverArtifactIdTest.java]: Moved it to the gradle related tests

**Review by kunli2 — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**